lib/stanford-mods/name.rb

```diff
@@ -45,22 +45,16 @@ def additional_authors_w_dates
         results
       end
 
-      COLLECTOR_ROLE_URI = 'http://id.loc.gov/vocabulary/relators/col'
-
       # @return Array of Strings, each containing the computed display value of a personal name
       #   except for the collector role (see mods gem nom_terminology for display value algorithm)
       # FIXME:  this is broken if there are multiple role codes and some of them are not marcrelator
       def non_collector_person_authors
         result = []
         @mods_ng_xml.personal_name.map do |n|
```

Comment:
I'm a little confused by this `map/each/result` business (especially the differences between this and in `collectors_with_date`). This seems like a perfect application for using some of the `Enumerable` methods:

```ruby
@mods_ng_xml.personal_name.select { |n| n.role.any? { |r| includes_marc_relator_collector_role?(r) } }
                                                    .map { |n| n.display_value_w_date }
```

This would also, presumably, stop duplicate name values for people with multiple roles.